chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -162,9 +162,10 @@ class ItemTreeManager {
 				enabledTreeIDs = [enabledTreeIDs];
 			}
 			filteredColumns = filteredColumns.filter((col) => {
-				return col.enabledTreeIDs?.includes("*")
+				const colEnabledTreeIDs = col.enabledTreeIDs || ["main"];
+				return colEnabledTreeIDs.includes("*")
 					|| enabledTreeIDs.includes("*")
-					|| enabledTreeIDs.find(treeID => (col.enabledTreeIDs || ["main"]).includes(treeID));
+					|| enabledTreeIDs.find(treeID => colEnabledTreeIDs.includes(treeID));
```

Comment:
This code is still confusing and inefficient.

- `column.enabledTreeIDs` should never be undefined. Assign it upon registration if not defined by client code.
- The check for `enabledTreeIDs.includes("*")` should be done once and then running this filter shouldn't even be necessary, just return like on L157.
- If a column registers "*" for enabledTreeIDs, it should be the only element (anything else discarded) and the comparison should be `colEnabledTreeIDs[0] == "*"`.
- I see `filteredColumns.filter(something.find(other.includes(id)))`. This needs to be unrolled, it's a triply-nested for loop rolled into a function comprehension.

Reply:
Ok, done